docs/diagram.md

```diff
@@ -0,0 +1,122 @@
+PipelineN --run downstream --> DS3[build-scripts/jobs/jdk<ver>/jdk<ver>-<os>-<arch>-<variant>]
+PipelineN --run downstream --> DS12[build-scripts/jobs/jdk<ver>/jdk<ver>-<os>-<arch>-<variant>]
+```
+
+## Mainflow logic of creation pipeline: openjdk*ver-pipeline
+
+```mermaid
+flowchart TD 
+Call[build/openjdk_pipeline.groovy] --load script--> 
+Load1[build/common/build_base_file.groovy]
+ 
+Call --load config--> Load2[build/jobs/configurations/jdk<ver>_pipeline_config.groovy]
+
+Call --sharedlib --> 
+Load3[Git repo: openjdk-jenkins-helper]
+
+Load1 --call_function--> Build[doBuild]
+Load2 --input--> Build
+Load3 --input--> Build
+
+Build --create_downstream_jobs --> 
+Done[Job: jdk<ver>/job/jdk<ver>-<os>-<arch>-<variant>]
+```
+
+## Mainflow logic of creation job: "build-scripts/job/utils/job/pipeline_jobs_generator_jdk*ver"
+
+```mermaid
+flowchart TD 
+build/regeneration/build_job_generator.groovy --load--> build/common/import_lib.groovy --load--> jobs/configurations/jdk*_pipeline_config.groovy --load--> jobs/configurations/jdk*.groovy --load--> common/config_regeneration.groovy --call_function--> regenerate --call_function--> makeJob --call--> createJob --> DSL[jobDSL: common/create_job_from_template.groovy] --set--> scm[Git clone: ci-jenkins-pipelines] --set--> properties[Permission/logRotator/...]--> parameters
+```
+
+## Mainflow logic of build job: jobs/jdk*ver/jdk*ver-*os-*arch-*variant
+
+```mermaid
+flowchart TD
+starter[kick_off_build.groovy] --load--> import[build/common/import_lib.groovy] --load--> Load1[build/common/openjdk_build_pipeline.groovy] --call_function--> Builder[build] 
+Builder --sharedlib --> 
+Load3[Git repo: openjdk-jenkins-helper] --> docker{build in docker}
+docker --yes: run--> dockerbuild[docker.build] --> sign
+docker --no: call_function--> CallbuildScript[buildScripts] --> sign{enableSigner} --yes:call--> sing[sign] --> testStage{enableTests}
+sign{enableSigner} --no:skip --> testStage --yes:call_function--> smoketest[runSmokeTests] --pass:run--> 
+smoke[Job: jobs/jdk<ver>/jdk<ver>-<os>-<arch>-<variant>_SmokeTests] --call_function--> Stage2[runAQATests]
+
+Stage2 --run_job--> sanity1[sanity.openjdk]
+Stage2 --run_job--> sanity2[sanity.system]
+Stage2 --run_job--> sanity3[sanity.perf]
+Stage2 --run_job--> sanity4[sanity.functional]
+Stage2 --run_job--> extended1[extended.system]
+Stage2 --run_job--> extended2[extended.functional]
+
+sanity1 --if:pass--> shouldInstaller
+sanity2 --if:pass--> shouldInstaller
+sanity3 --if:pass--> shouldInstaller
+sanity4 --if:pass--> shouldInstaller
+extended1 --if:pass--> shouldInstaller
+extended2 --if:pass--> shouldInstaller
+shouldInstaller --> install{enableInstaller} --yes:call_function--> bI[buildInstaller] --> sI[signInstaller]
+ ```
+
+ ```mermaid
+flowchart TD 
+CallbuildScript[function: buildScripts] --git_clone--> checkout[temurin-build] --call_script -->
+script[sh:script build-farm/make-adopt-build-farm.sh] --call_function--> meta[writeMetadata] --call_function -->
```

Comment:
probably needs a bit of extra detail between make-adopt-build-farm.sh and writeMetadata, ie. the detail on how build.sh gets run and does the actual openjdk "build"
Also, add CycloneDX SBOM creation after writeMetadata

Reply:
added some more for the make-adopt-build-farm, buildCyclonedxLib and generateSBoM

Reply:
@zdtsw the diagram for build.sh is not quite right, the two options MAKE_EXPLODED and ASSEMBLE_EXPLODED_IMAGE, are specific to MacOS only, as they are allow Apple "signing" of the jmods, prior to packaging. So it does it in 2 phases of invocation of build.sh, first with  --make-exploded-image, it then does the Mac signing, it then does --assemble-exploded-image.
So strictly speaking, the diagram needs an if "MacOS" then do the 2 phase invocation of sbin/build.sh.
otherwise both those options are "false", and the script simply does the standard build sequence:
- "OpenJDK Build"
- Create Release file and metadata
- If --create-sbom then create the SBOM
- Create archive tarballs

Reply:
Thanks! I will update with more logic checks

Reply:
@andrew-m-leonard did updates for the "macos" checks mainly in "Breakdown logic of build script: openjdk_build_pipeline.groovy" section

Reply:
looks good